### ISSUE TITLE
CONNECT_CR_SRVR_HELLO:wrong version

### ISSUE THREAD

**Issue by jimkoen** [yhirose/cpp-httplib#1115]
I am getting a CONNECT_CR_SRVR_HELLO:wrong version error when trying to connect to an instance of httplib::SSLServer.

It looks like it is unable to find or serve the respective cert and keyfile.

The certs the server uses are self signed, but are otherwise valid. I am unable to debug the issue any further, except provide debug output from SSL.

The code that initialises the Server:
```c++
Http::Server::THttpServerInstance::THttpServerInstance() : mHttpLibServerInstance{Application::Settings.SSLCertPath.c_str(), Application::Settings.SSLKeyPath.c_str()} {
    mHttpLibServerInstance.Get("/", [](const httplib::Request&, httplib::Response& res) {
        res.set_content("Hello World!", "text/html");
    });

    Start();
}
void Http::Server::THttpServerInstance::operator()() {
    mHttpLibServerInstance.listen("0.0.0.0", 23417);
}
```
Note that `THttpServerInstance` inherits from another class, of which `Start()` is a member function. `Start()` takes the `operator()`of the current class and starts it in another thread.

When putting `mHttpLibServerInstance.listen("0.0.0.0", 23417);` directly into `THttpServerInstance`'s constructor, the server will handle TLS requests just fine..

I do not understand why the server is unable to serve/find the keyfile/certfile when running in another thread.
The application is mostly threadsafe, I cannot spot any immediate issue. Regardless from where listen is called,
the paths seem to always be initialised.

Output of `openssl s_client -connect localhost:23417 -servername localhost`:
```
CONNECTED(00000005)
4463154860:error:1400410B:SSL routines:CONNECT_CR_SRVR_HELLO:wrong version number:/System/Volumes/Data/SWE/macOS/BuildRoots/b8ff8433dc/Library/Caches/com.apple.xbs/Sources/libressl/libressl-75/libressl-2.8/ssl/ssl_pkt.c:386:
---
no peer certificate available
---
No client certificate CA names sent
---
SSL handshake has read 5 bytes and written 0 bytes
---
New, (NONE), Cipher is (NONE)
Secure Renegotiation IS NOT supported
Compression: NONE
Expansion: NONE
No ALPN negotiated
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    Start Time: 1638595370
    Timeout   : 7200 (sec)
    Verify return code: 0 (ok)
---
```

**Comment by jimkoen** [comment]
Nvm, this is a fundamental design issue, due to the library not being thread safe. The solution was to put the SSLServer instance inside a shared_ptr and initialize the instance + call `listen()` inside `Start()`, so that all calls happen in a separate thread.

tl;dr: Avoid configuring httlib Server instances across different threads.

**Comment by yhirose** [comment]
@jimkoen, thank you for your feedback. There are actually some SSL server examples that calls `listen` method in a separate thread, and they are working fine on my machine and GitHub Actions. Here is one of them:

https://github.com/yhirose/cpp-httplib/blob/824e7682e4d95e1bb21e501731eb2b6bb23033d2/test/test.cc#L4196-L4219

As you see in httplib.h, I tried to protect `SSL_CTX *ctx_` with `std::mutex ctx_mutex_` as much as I can. So if your problem is a really 'thread-safe' issue, please let me know where such a problem is caused. Thanks a lot for your help!
